db/records/operations/insert.py

```diff
@@ -32,9 +33,14 @@ def insert_record_or_records(table, engine, record_data):
     return None
 
 
-def insert_records_from_json(table, engine, json_filepath):
+def insert_records_from_json(table, engine, json_filepath, column_names):
```

Comment:
Add a docstring. Include explanation for why `column_names` is necessary. Include explanation for why and how `json_normalize` is being invoked (what's `max_level` and why is it `0`). Include explanation of the general algorithm and intent.

Reply:
Done. Thanks!

Reply:
Include a summary and explation of the algorithm. You normalize into a dataframe, then convert to a JSON string, then to Python, ending up with a sequence of rows where each row is dict-like, in which every key-value pair is a column, then you take those columns and if they are a dict or a list, you serialize them back into JSON. That takes some time to figure out and the reason for having to do some of these things is not necessarily immediately obvious. Help the reader out with a summary and an explanation.

Reply:
Presume that the reader is not me or anyone that you've discussed JSON importing with.

Reply:
Hi Dom, I have added the explanation now. PTAL. Thanks!

Reply:
Awesome!